src/components/form/weekday.tsx

```diff
@@ -0,0 +1,70 @@
+import { capitalize } from "lodash";
+import { UseFormRegisterReturn } from "react-hook-form";
+import { HTMLProps, forwardRef, useState } from "react";
+import { enumKeys, Weekday, shortenWeekday } from "../../utils";
+
+type Props = UseFormRegisterReturn<string> &
+  HTMLProps<HTMLInputElement> & {
+    labelText?: string;
+    hintText?: string;
+    feedback?: string;
+    className?: string;
+  };
+
+export const WeekdaySelector = forwardRef<HTMLInputElement, Props>(
+  ({ labelText, hintText, feedback, className, ...rest }: Props, ref) => {
+    const [currentSelectedDays, setCurrentSelectedDays] = useState<string[]>(
+      [],
+    );
+    const [uniqueId] = useState(
+      () => `weekday-${Date.now()}-${Math.floor(Math.random() * 1000)}`,
+    );
+
+    const handleCheck = (day: Weekday) => {
+      if (currentSelectedDays.includes(day)) {
+        setCurrentSelectedDays(currentSelectedDays.filter((d) => d != day));
+      } else {
+        setCurrentSelectedDays([...currentSelectedDays, day]);
+      }
+    };
+
+    return (
+      <div className={`max-w-pageContent mx-3 relative font-text ${className}`}>
+        <label className="">{labelText}</label>
+        <div className="flex justify-between mt-3">
+          {enumKeys(Weekday).map((day) => (
```

Comment:
Slight change here, defined an Enum for the days of the week so we can reference the values elsewhere in the code instead of having an array of strings defined here 👍